python/mxnet/gluon/contrib/data/audio/datasets.py

```diff
@@ -0,0 +1,171 @@
+    def __init__(self, root, has_csv=False, train_csv=None, file_format='.wav', skip_rows=0):
+        self._root = os.path.expanduser(root)
+        self._exts = ['.wav']
+        self._format = file_format
+        self._has_csv = has_csv
+        self._train_csv = train_csv
+        self._list_audio_files(self._root, skip_rows=skip_rows)
+
+
+    def _list_audio_files(self, root, skip_rows=0):
+        """
+            Populates synsets - a map of index to label for the data items.
+            Populates the data in the dataset, making tuples of (data, label)
+        """
+        self.synsets = []
+        self.items = []
+        if not self._has_csv:
+            for folder in sorted(os.listdir(root)):
+                path = os.path.join(root, folder)
+                if not os.path.isdir(path):
+                    warnings.warn('Ignoring %s, which is not a directory.'%path, stacklevel=3)
+                    continue
+                label = len(self.synsets)
+                self.synsets.append(folder)
+                for filename in sorted(os.listdir(path)):
+                    file_name = os.path.join(path, filename)
+                    ext = os.path.splitext(file_name)[1]
+                    if ext.lower() not in self._exts:
+                        warnings.warn('Ignoring %s of type %s. Only support %s'%(filename, ext, ', '.join(self._exts)))
+                        continue
+                    self.items.append((file_name, label))
+        else:
+            data_tmp = []
+            label_tmp = []
+            skipped_rows = 0
+            with open(self._train_csv, "r") as traincsv:
+                for line in traincsv:
+                    skipped_rows = skipped_rows + 1
+                    if skipped_rows <= skip_rows:
+                        continue
+                    filename = os.path.join(root, line.split(",")[0])
+                    label = line.split(",")[1].strip()
+                    if label not in self.synsets:
+                        self.synsets.append(label)
+                    data_tmp.append(os.path.join(self._root, line.split(",")[0]))
+                    label_tmp.append(self.synsets.index(label))
+
+            #Generating the synset.txt file now
+            with open("./synset.txt", "w") as synsets_file:
+                for item in self.synsets:
+                    synsets_file.write(item+os.linesep)
+            print("Synsets is generated  as synset.txt")
+
+            self._label = nd.array(label_tmp)
+            for i, _ in enumerate(data_tmp):
+                if self._format not in data_tmp[i]:
+                    self.items.append((data_tmp[i]+self._format, self._label[i]))
+
+    def __getitem__(self, idx):
+        """
+            Retrieve the item (data, label) stored at idx in items
+        """
+        filename = self.items[idx][0]
+        label = self.items[idx][1]
+
+        if librosa is not None:
+            X1, _ = librosa.load(filename, res_type='kaiser_fast')
```

Comment:
why kaiser_fast, why not kaiser_best?

Reply:
'kaiser_fast' is faster than 'kaiser_best'
https://librosa.github.io/librosa/generated/librosa.core.load.html
We can make it as a keyword argument if need be.